app/models/service.rb

```diff
@@ -179,12 +179,16 @@ def queue_group_action(action, group_idx, direction, deliver_delay)
       :args        => [action, group_idx, direction]
     }
     nh[:deliver_on] = deliver_delay.seconds.from_now.utc if deliver_delay > 0
-    first_vm = vms.first
-    nh[:zone] = first_vm.ext_management_system.zone.name unless first_vm.nil?
+    nh[:zone] = my_zone
```

Comment:
We only want to define nh[:zone] if my_zone is not nil.  The code in app/models/zone.rb's determine_queue_zone only checks if the :zone key exists.  Thanks.